CMakeLists.txt

```diff
@@ -1199,13 +1199,17 @@ zephyr_compile_options(
   $<$<COMPILE_LANGUAGE:C>:-std=${CSTD}>
 )
 
+if(CONFIG_ARC)
 configure_file(
      $ENV{ZEPHYR_BASE}/include/arch/arc/v2/app_data_alignment.ld
      ${PROJECT_BINARY_DIR}/include/generated/app_data_alignment.ld)
+endif()
 
+if(CONFIG_ARM)
 configure_file(
      $ENV{ZEPHYR_BASE}/include/arch/arm/cortex_m/scripts/app_smem.ld
      ${PROJECT_BINARY_DIR}/include/generated/app_smem.ld)
+endif()
```

Comment:
IIRC one of these scripts was re-used by multiple architectures, or did we fix that @ioannisg ?

Reply:
The first change is fine; app_data_alignment is not used in ARM.

Reply:
app_smem_ld is also used by ARC , AFAIK. So we will have a problem if we do as the patch suggests, right @SebastianBoe ?

Reply:
Yep. That seems to be correct.

```
sebo@mach:~/ncs/zephyr$ ack app_smem.ld ~/ncs/zephyr/
ack app_smem.ld ~/ncs/zephyr/
/home/sebo/ncs/zephyr/include/arch/arm/cortex_m/scripts/linker.ld
243:		#include <app_smem.ld>

/home/sebo/ncs/zephyr/include/arch/arc/v2/linker.ld
133:		#include <app_smem.ld>

/home/sebo/ncs/zephyr/CMakeLists.txt
1024:     $ENV{ZEPHYR_BASE}/include/arch/arm/cortex_m/scripts/app_smem.ld
1025:     ${PROJECT_BINARY_DIR}/include/generated/app_smem.ld)
1030:        set(APP_SMEM_LD "${PROJECT_BINARY_DIR}/include/generated/app_smem.ld")

```

Reply:
But it might be a moot point if @andrewboie is going to rewrite this ...

Reply:
> app_smem_ld is also used by ARC , AFAIK. So we will have a problem if we do as the patch suggests, right @SebastianBoe ?

Then we need to move those from include/<ARCH>/ and put them in some common place.